src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/Step.kt

```diff
@@ -26,11 +26,9 @@ abstract class Step(val name: String, val managedIndexMetaData: ManagedIndexMeta
     abstract fun getUpdatedManagedIndexMetaData(currentMetaData: ManagedIndexMetaData): ManagedIndexMetaData
 
     fun getStepStartTime(): Instant {
-        val startTimeMillis = managedIndexMetaData.stepStartTime
-        if (managedIndexMetaData.step != name || startTimeMillis == null) {
+        if (managedIndexMetaData.stepMetaData == null) {
```

Comment:
Do we still need a check for the step name here?

Reply:
no, StepMetaData is a data class that requires non-null Step name and StartTime. So checking if StepMetaData is not null is enough indication that the step is on first execution and can return `Instant.now()`

Reply:
We need to check if the step name is the same or else we need to return new start time.

I will make the change.